test/functional/feature_avoidreuse.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the avoidreuse feature."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+    connect_nodes_bi,
+)
+
+# TODO: Copied from wallet_groups.py -- should perhaps move into util.py
+def assert_approx(v, vexp, vspan=0.00001):
+    if v < vexp - vspan:
+        raise AssertionError("%s < [%s..%s]" % (str(v), str(vexp - vspan), str(vexp + vspan)))
+    if v > vexp + vspan:
+        raise AssertionError("%s > [%s..%s]" % (str(v), str(vexp - vspan), str(vexp + vspan)))
+
+class AvoidReuseTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+        self.extra_args = [[], ['-avoidreuse=1']]
+
+    def reset_balance(self, node, discardaddr):
+        '''
+        Throw away all owned coins by the node so it gets a balance of 0.
+        '''
+        balance = node.getbalance(include_dirty=True)
+        if balance > 0.5:
+            node.sendtoaddress(address=discardaddr, amount=balance, subtractfeefromamount=True, allow_dirty=True)
+
+    def run_test(self):
+        '''
+        Set up initial chain and run tests defined below
+        '''
+
+        self.test_persistence()
+
+        self.nodes[0].generate(110)
+        self.sync_all()
+        self.reset_balance(self.nodes[1], self.nodes[0].getnewaddress())
+        self.test_fund_send_fund_send()
+        self.reset_balance(self.nodes[1], self.nodes[0].getnewaddress())
+        self.test_fund_send_fund_senddirty()
+
+    def test_persistence(self):
+        '''
+        Test that wallet files persist the -avoidreuse flag.
+        '''
+        # TODO: Using -avoidreuse as startup option SHOULD cause the wallet to
+        # TODO: have the avoidreuse flag set already, but this is not the case
+        # Configure node 1 to use avoidreuse
+        self.nodes[1].setwalletflag('avoidreuse')
+
+        # Flags should be node1.avoidreuse=false, node2.avoidreuse=true
+        assert_equal(self.nodes[0].getwalletflags('avoidreuse'), False)
+        assert_equal(self.nodes[1].getwalletflags('avoidreuse'), True)
+
+        # Stop and restart node 1 without -avoidreuse flag
+        self.stop_node(1)
+        self.start_node(1)
+        connect_nodes_bi(self.nodes, 0, 1)
+
+        # Flags should still be node1.avoidreuse=false, node2.avoidreuse=true
+        assert_equal(self.nodes[0].getwalletflags('avoidreuse'), False)
+        assert_equal(self.nodes[1].getwalletflags('avoidreuse'), True)
+
+    def test_fund_send_fund_send(self):
+        '''
+        Test the simple case where [1] generates a new address A, then
+        [0] sends 10 BTC to A.
+        [1] spends 5 BTC from A. (leaving roughly 4 BTC useable)
```

Comment:
Typo found by `codespell`: useable

Reply:
> Usable can also be spelled useable, with an e in the middle: both are correct.

https://english.stackexchange.com/questions/69709/is-useable-preferred-in-certain-regions-or-just-an-alternate-spelling-of-usa